homeassistant/components/knx/device_trigger.py

```diff
@@ -63,17 +78,39 @@ async def async_get_trigger_capabilities(
         selector.SelectOptionDict(value=ga.address, label=f"{ga.address} - {ga.name}")
         for ga in project.group_addresses.values()
     ]
+    is_legacy_config = not all(
+        key in config
+        for key in (
+            CONF_KNX_GROUP_VALUE_WRITE,
+            CONF_KNX_GROUP_VALUE_RESPONSE,
+            CONF_KNX_GROUP_VALUE_READ,
+            CONF_KNX_INCOMING,
+            CONF_KNX_OUTGOING,
+        )
+    )
+    default_selectors = {
+        vol.Optional(CONF_KNX_DESTINATION): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                mode=selector.SelectSelectorMode.DROPDOWN,
+                multiple=True,
+                custom_value=True,
+                options=options,
+            ),
+        ),
+    }
+    additional_selectors = {
+        # need to be set in async_get_triggers to have proper default values
+        vol.Required(CONF_KNX_GROUP_VALUE_WRITE): selector.BooleanSelector(),
+        vol.Required(CONF_KNX_GROUP_VALUE_RESPONSE): selector.BooleanSelector(),
+        vol.Required(CONF_KNX_GROUP_VALUE_READ): selector.BooleanSelector(),
+        vol.Required(CONF_KNX_INCOMING): selector.BooleanSelector(),
+        vol.Required(CONF_KNX_OUTGOING): selector.BooleanSelector(),
```

Comment:
Does it not work if you add default values to these?

Reply:
The default values have to be set in `async_get_triggers`, otherwise the resulting yaml looks like this (even with default values provided in `async_get_trigger_capabilities`).
```yaml
platform: device
domain: knx
device_id: 0ef7e391fb6fa4e19f8e9048958ee994
type: telegram
```
Even worse, if the `default=True`, this will be applied, but all the switches in UI are "off" so it doesn't represent what will be created at all.

The current implementation results in correct UI switches (on by default) and this yaml
```yaml
platform: device
domain: knx
device_id: 0ef7e391fb6fa4e19f8e9048958ee994
type: telegram
group_value_write: true
group_value_response: true
group_value_read: true
incoming: true
outgoing: true
```

Reply:
I tried using a `ConstantSelector` too, but this doesn't show a checkbox (like shown in https://www.home-assistant.io/docs/blueprint/selectors/#constant-selector) so it isn't de-selectable.